finality-aleph/src/network/clique/protocols/v0/mod.rs

```diff
@@ -123,14 +138,45 @@ pub async fn incoming<SK: SecretKey, D: Data, S: Splittable>(
     }
 }
 
+pub async fn handle_authorization<SK: SecretKey>(
+    authorization_requests_sender: mpsc::UnboundedSender<(SK::PublicKey, oneshot::Sender<bool>)>,
+    public_key: SK::PublicKey,
+) -> Result<bool, ()> {
+    let (sender, receiver) = oneshot::channel();
+    if let Err(err) = authorization_requests_sender.unbounded_send((public_key.clone(), sender)) {
+        warn!(
```

Comment:
Related to the previous point – if you return meaningful errors from here then you don't have to `log` inside this function, simplifying the code quite a bit. Although maybe that's just my preference for "log as high up the stack as possible"?

Reply:
changed